ykutil/src/addr.rs

```diff
@@ -5,16 +5,66 @@ use libc::{self, c_void, Dl_info};
 use phdrs::objects;
 use std::mem::MaybeUninit;
 use std::{
-    convert::TryFrom,
+    convert::{From, TryFrom},
     ffi::CStr,
     path::{Path, PathBuf},
-    ptr::null,
 };
 
-pub fn dladdr(vaddr: usize) -> Result<Dl_info, ()> {
+/// A Rust wrapper around `libc::Dl_info` using FFI types.
+///
+/// The strings inside are handed out by the loader and can (for now, since we don't support
+/// dlclose) be considered of static lifetime.
+#[derive(Debug)]
+pub struct DLInfo {
+    dli_fname: Option<&'static CStr>,
+    dli_fbase: usize,
+    dli_sname: Option<&'static CStr>,
+    dli_saddr: usize,
+}
+
+impl From<Dl_info> for DLInfo {
+    fn from(dli: Dl_info) -> Self {
+        let dli_fname = if !dli.dli_fname.is_null() {
```

Comment:
Isn't this something like `dli.dli_fname.map(|x| CStr::from_ptr(x))` in disguise?

Reply:
I can't see any `ptr::map()` in the std lib... What am I missing?

`dli.dli_fname` is a [`*const c_char`](https://docs.rs/libc/0.2.139/libc/struct.Dl_info.html).

Reply:
Oh, oops!